components/index_aws_opensearch/src/main.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Dict
+
+import boto3
+import dask.dataframe as dd
+from fondant.component import DaskWriteComponent
+from opensearchpy import AWSV4SignerAuth, OpenSearch, RequestsHttpConnection
+
+
+class IndexAWSOpenSearchComponent(DaskWriteComponent):
+    def __init__(
+        self,
+        host: str,
+        region: str,
+        index_name: str,
+        index_body: Dict[str, Any],
+        port: int = 443,
```

Comment:
I would not include the default types here for the optional argument since you already define them in the spec and it can be misleading (they are not actually used here it will always default to the ones defined in the spec). You can instead define their types as optional just so that it's clear that they do have default values and they don't need to be explicitly defined. Similar to [here](https://github.com/ml6team/fondant/blob/main/components/load_from_hf_hub/src/main.py)